### ISSUE TITLE
Question about S3 Staticfile support

### ISSUE THREAD

**Issue by benjiboi214** [tendenci/tendenci#979]
Hi Tendenci Devs!

Love your work on this project so far. Really awesome to see how in depth the functionality of this platform is.

I am currently trying to self host this for an association I work with, and have question about the extent of S3 support provided in the current release.

I can see in two files that there are methods for running some S3 operations. 

First there is `tendenci/apps/base/management/commands/s3_replace_static_urls.py` which seems to be a command to assist with migration to S3 on an active deployment.
Is that correct?

Second there are a bunch of static and media utilities in `tendenci/libs/boto_s3/utils.py`.
I just wanted to understand if I might be able to help port these to boto3 if these are required to support S3 as a static and media host? It seems the methods here are mostly used for the theme editor. 

It seems django-storages should take care of most of this functionality from a static perspective? So is it just the theme editor that is not yet currently migrated to boto3?

Anyway, any direction you might be able to give me would be greatly appreciated so i know where to focus my effort helpfully!
Thanks.

**Comment by jennyq** [comment]
Hi @benjiboi214.

Honestly, I'm not sure if the current version of Tendenci works with S3 storage or not. We did use it in the earlier stage with Heroku for a very short period of time, but the codebase has been changed dramatically ever since. I remember it was a lot of hassles to copy files from and to S3 because django doesn't serve templates remotely, and it is not possible to load TinyMCE crossdomain. 

You can try for the media files first by following the instructions from django-storages https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html to set the settings in your conf/settings.py, and set the `S3_ROOT_URL` setting. If that works, you can help to port those utility functions in `libs/boto_s3/utils.py` to boto3. The command `s3_replace_static_urls` is no longer needed I believe. 

Thanks.

**Comment by benjiboi214** [comment]
Yeah I think the themes app is throwing a massive spanner in the works as far as I can tell.

I tried a few permutations of the settings in django storages, as well as some manual exfiltration, and it seems that there is quite a few different approaches depending on whether the static content is from a theme, or gathered from the app static folder.
Even then, there seems to be some variation depending on whether the content is a base theme or not.

At this point I'm not sure I am close enough to the project to dive in to making this work just yet.

I'm guessing the current production deployment is the [standard django approach](https://docs.djangoproject.com/en/3.1/howto/static-files/deployment/) - essentially just gather the static content to `STATIC_ROOT` and then you reverse proxy to that directory?

**Comment by jennyq** [comment]
> I'm guessing the current production deployment is the [standard django approach](https://docs.djangoproject.com/en/3.1/howto/static-files/deployment/) - essentially just gather the static content to `STATIC_ROOT` and then you reverse proxy to that directory?

That is correct. 

The static files in the themes directory are also collected to the `STATIC_ROOT` with Django's `collectstatic` command. The `static` tag in Tendenci works the same as the standard Django `{% static %}` behavior except it makes sure the static files in the current theme have the higher precedence being served than the rest of the files with the same name. https://github.com/tendenci/tendenci/blob/master/tendenci/apps/theme/templatetags/static.py#L30
